urllib3/gae_connectionpool.py

```diff
@@ -0,0 +1,96 @@
+from .connectionpool import (
+    ConnectionPool,
+    _Default,
+    get_host)
+
+from .exceptions import (
+    ProtocolError,
+    HostChangedError,
+    LocationValueError,
+    MaxRetryError,
+    SSLError,
+    TimeoutError,
+    InsecureRequestWarning,
+)
+
+from .packages.six import BytesIO
+from .request import RequestMethods
+from .response import HTTPResponse
+from .util.timeout import Timeout
+from .util.url import Url
+
+try:
+    from google.appengine.api import urlfetch
+except ImportError:
+    urlfetch = None
+
+
+class GAEConnectionPool(ConnectionPool, RequestMethods):
+
+    def __init__(self, host, port=None, scheme='http', timeout=5, headers=None,
+                 retries=None, validate_certificate=False):
+
+        if not urlfetch:
+            raise EnvironmentError(
+                "URLFetch is not available in this environment.")
+
+        ConnectionPool.__init__(self, host, port)
+        RequestMethods.__init__(self, headers)
+
+        # URLFetch doesn't support fine-grained control over timeouts.
+        if isinstance(timeout, Timeout):
+            timeout = timeout.total
```

Comment:
Ah I misunderstood what you said. All the ways to set a timeout: https://urllib3.readthedocs.org/en/latest/#timeout

So maybe something like...

```python
if isinstance(timeout, Timeout) and not (timeout.read is timeout.connect is Timeout.DEFAULT_TIMEOUT):
    warnings.warn("URLFetch does not support granular timeout settings, reverting to total timeout.", GAEPlatformWarning)
```